chainer/training/extensions/graph_report.py

```diff
@@ -0,0 +1,141 @@
+from chainer import reporter
+import chainer.serializer as serializer_module
+from chainer.training import extension
+import chainer.training.trigger as trigger_module
+import json
+import numpy as np
+from os import path
+import six
+
+try:
+    import matplotlib
+
+    matplotlib.use('Agg')
+    from matplotlib import pyplot as plot
+
+    _available = True
+
+except ImportError:
+    _available = False
+
+
+def _check_available():
+    if not _available:
+        msg = '''matplotlib is not installed on your environment.
+Please install matplotlib to plot figure.
+
+  $ pip install matplotlib'''
+        raise RuntimeError(msg)
+
+
+class GraphReport(extension.Extension):
+
+    def __init__(self, y_keys, x_key='iteration', trigger=(1, 'epoch'),
+                 postprocess=None, file_name='graph.png'):
+        """Trainer extension to output graph.
```

Comment:
Could you move this docstring to the right after the class name (outside the `__init__` function)?